packages/emitter-framework/src/core/components/output.tsx

```diff
@@ -0,0 +1,18 @@
+import * as ay from "@alloy-js/core";
+import { Program } from "@typespec/compiler";
+import { TspContext } from "../context/tsp-context.js";
+
+export interface OutputProps extends ay.OutputProps {
+  /**
+   * TypeSpec program.
+   */
+  program: Program;
+}
+
+export function Output({ program, ...rest }: OutputProps) {
```

Comment:
```suggestion
export function Output(props:Readonly<OutputProps>) {
```
talked in separate chat that alloy component should follow this as mutating  or even deconstructing the props breaks things.
- Mutating props breaks if the props used a computed value
- Deconstructing or spreading props make the props loose their reactivity

Reply:
This really surprised me - thanks for sharing - updated!